### ISSUE TITLE
--ignore-branches-older-than option throwing error on some organizations

### ISSUE THREAD

**Issue by KarahanGuner** [punk-security/secret-magpie-cli#49]
--ignore-branches-older-than option throws an error on some organizations. Example:
```
kali@kali:~$ sudo docker run punksecurity/secret-magpie github --org MetaMask --pat [github token] --ignore-branches-older-than 2018-01-01
          ____              __   _____                      _ __       
         / __ \__  ______  / /__/ ___/___  _______  _______(_) /___  __
        / /_/ / / / / __ \/ //_/\__ \/ _ \/ ___/ / / / ___/ / __/ / / /
       / ____/ /_/ / / / / ,<  ___/ /  __/ /__/ /_/ / /  / / /_/ /_/ / 
      /_/    \__,_/_/ /_/_/|_|/____/\___/\___/\__,_/_/  /_/\__/\__, /  
                                             PRESENTS         /____/  
                              Secret-Magpie ✨

      Scan all your github/bitbucket repos from one tool, with multiple tools!
        
Traceback (most recent call last): Total secret detections: 0 |
  File "/app/main.py", line 52, in <module>
    for result_batch in results:
  File "/usr/local/lib/python3.10/multiprocessing/pool.py", line 873, in next
    raise value
  File "/usr/local/lib/python3.10/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/app/tasks.py", line 98, in process_repo
    branches = get_branches(
  File "/app/tasks.py", line 58, in get_branches
    branches = list(
  File "/app/tasks.py", line 60, in <lambda>
    lambda branch: r.commit(branch).committed_date >= threshold_date,
  File "/opt/venv/lib/python3.10/site-packages/git/repo/base.py", line 606, in commit
    return self.rev_parse(str(rev) + "^0")
  File "/opt/venv/lib/python3.10/site-packages/git/repo/fun.py", line 247, in rev_parse
    ref = cast("Reference", name_to_object(repo, rev[:start], return_ref=True))
  File "/opt/venv/lib/python3.10/site-packages/git/repo/fun.py", line 178, in name_to_object
    raise BadObject("Couldn't find reference named %r" % name)
gitdb.exc.BadObject: <exception str() failed>
```

I tried with 2018-01-01T00:00:00+00:00 as a time input, but it gave the same error.

**Comment by SimonGurney** [comment]
Hmmm... not sure why you don't instantly see the banner.  Can you run it with "-t" on the docker args and see if you get the banner immediately?

**Comment by SimonGurney** [comment]
This was because we didnt handle the exception to the last commit lookup, which failed for some forks.  Fixed in #51
